### ISSUE TITLE
shallowEqual always returns false

### ISSUE THREAD

**Issue by paztis** [davej/js-redux#2]
HI

In shallowEqual, on first line you test
`if (typeof obj !== 'object')`
instead of 
`if (typeof objA !== 'object')`

Ad in you code obj var doesn't exists, typeof obj is undefined, and your shallowEqual always returns false.
It means we render and re-render all the time even if nothing has changed.

A workaround can be to define window.obj = {} somewhere to make you code work, but this will make the primitiveType not work.

Code fixed: 
```
export function shallowEqual(objA, objB) {
  if (typeof objA !== 'object') return objA === objB;
  if (objA === objB) return true;

  const keysA = Object.keys(objA);
  const keysB = Object.keys(objB);

  if (keysA.length !== keysB.length) return false;

  // Test for A's keys different from B.
  const hasOwn = Object.prototype.hasOwnProperty;
  for (let i = 0; i < keysA.length; i++) {
    if (!hasOwn.call(objB, keysA[i]) || objA[keysA[i]] !== objB[keysA[i]]) {
      return false;
    }
  }

  return true;
}
```

**Comment by davej** [comment]
Great catch, not sure how I missed that. Would you be able to do a PR?

**Comment by paztis** [comment]
Hi

Done

**Comment by davej** [comment]
Thanks, published in `v1.0.2`.
